### ISSUE TITLE
[Feature Request] Script hook on view refresh cycle.

### ISSUE THREAD

**Issue by RimBlock** [swar/Swar-Chia-Plot-Manager#1080]
Firstly thanks to Swar and the other contributers for this great system.

There are a number items that people would like implemented that have not made it in yet.

**i.e.** 
- Job stall check.
- Disk space checks and drive activation / deactivation depending on state.
- Post job plot moving.
- General system heathchecks.

One quick way to help people with this maybe to add the ablilty to run a script on each "view" cycle.   Frequency is controlled by view refresh timer.

**Advantages**
- People can run healthchechs etc without needing to merge code with the project.
- Saves people the need to play with Cron or MS Scheduler etc and to keep everything in the Swar bundle.
- People can also share scripts and the most popular ones could be evaluated for a future feature addition.
- Doesn't limit script runs to only after a plot has finished (although that hook will be great to have when it is merged and released).

**Examples of scripts for constand checks.**
-   Run a script that check to see if a jobs logfile has updated in the last 5 minutes.  If not, send a notification, scape the jobs PID and polt ID, kill the job, remove the PlotID's tmp files.
-   Run a script that checks if a disk is filling up, amend the config.yaml and # that location out.
-  Implement tmp and destination location rotation.   Have pools of tmp / dest paths (one for SSD / one for NVME, one for HDD) and amend the config.yaml with the next in the pool as one is used for a new plot process.
- General system health checks and alerting

**Comment by creamsoup** [comment]
for monitoring and health checking:
why don't you run a monitoring and health checking tools? if you don't want to mess with your own, there is a [chiamon](https://github.com/retzkek/chiamon) for you. also, swar plot manager is exposing limited set of metrics via prometheus. 

btw, i like the idea to have a "Job stall check" (I implemented this in my own plot manager which I can open source in the future =p still under development).

**Comment by RimBlock** [comment]
Yep, I am working on an ELK setup for that in my own setup.  Was thinking more for others.  Thanks for the chiamon link.
